testsuite/python/unittest_decorators.py

```diff
@@ -50,7 +51,19 @@ def skipIfMissingModules(*args):
 
 def skipIfMissingGPU():
     """Unittest skipIf decorator for missing GPU."""
-
     if not espressomd.gpu_available():
         return unittest.skip("Skipping test: no GPU available")
     return _id
+
+
+def skipIfUnmetModuleVersionRequirement(module, version_requirement):
+    """Unittest skipIf decorator for unmet module version requirement."""
+    try:
+        _module = importlib.import_module(module)
+    except ImportError:
+        return skipIfMissingModules(module)
+    if not setuptools.version.pkg_resources.packaging.specifiers.SpecifierSet(
+            version_requirement).contains(_module.__version__):
```

Comment:
This will fail on modules that don't provide a `__version__` variable, e.g. `ast`, but I can't see a workaround. Adding an extra  guard `assert hasattr(_module, '__version__')` wouldn't help, as it generates the same error message. We (core team) just need to be aware of that loophole when reviewing PRs that include this fixture.

Reply:
i think we can live with that